src/nvim/os/env.c

```diff
@@ -134,6 +157,35 @@ void init_homedir(void)
   if (var != NULL && *var == NUL)       /* empty is same as not set */
     var = NULL;
 
+#ifdef WIN32
+  /*
+   * Typically, $HOME is not defined on Windows, unless the user has
+   * specifically defined it for Vim's sake.  However, on Windows NT
+   * platforms, $HOMEDRIVE and $HOMEPATH are automatically defined for
```

Comment:
Not sure if you've touched this code already, as opposed to just moving it, but the style is off in a few places (braces position, `homedrive/homepath` could be declared inline.